pkg/splunk/enterprise/util.go

```diff
@@ -2272,3 +2272,49 @@ func getApplicablePodNameForK8Probes(cr splcommon.MetaObject, ordinalIdx int32)
 	}
 	return fmt.Sprintf("splunk-%s-%s-%d", cr.GetName(), podType, ordinalIdx)
 }
+
+// getCurrentImage gets the image of the statefulset, returns the image, and error if something goes wrong
+func getCurrentImage(ctx context.Context, c splcommon.ControllerClient, cr splcommon.MetaObject, instanceType InstanceType) (string, error) {
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetSplunkStatefulsetName(instanceType, cr.GetName()),
+	}
+	statefulSet := &appsv1.StatefulSet{}
+	err := c.Get(ctx, namespacedName, statefulSet)
+	if err != nil {
+		return "", err
+	}
+
+	if statefulSet.Spec.Template.Spec.Containers == nil {
```

Comment:
Can we make this something like:

```
if len(statefulSet.Spec.Template.Spec.Containers) > 0 {
      return statefulSet.Spec.Template.Spec.Containers[0].Image
}

return "", <error stating that we couldn't find an image>
```
We need to catch the error condition where the image is not present in the spec to avoid a silent ignore of it.

Reply:
I think throwing an error could cause an issue during the initial deployment. Is it possible that the stateful set is initialized but the container is still in creation? If yes, this would stop the instance reconciliation/set the phase of the instance to `Error` till the container is fully up. The reconcile will not get stuck though because of the multiple attempts that take place.

Reply:
I think throwing an error could cause an issue during the initial deployment. Is it possible that the stateful set is initialized but the container is still in creation? If yes, this would stop the instance reconciliation/set the phase of the instance to `Error` till the container is fully up. The reconcile will not get stuck though because of the multiple attempts that take place.

Reply:
We are reading the statefulSet spec from etcd and not checking if the container is actually UP so we should be fine.